### PR TITLE
Add encryptionkey.GetCurrentState() implementation

### PR DESCRIPTION
**Pull request by tuommaki**
Simple implementation that looks for secret named after clusterID.

### CODE REVIEW

File: service/kvmclusterconfig/v1/resource/encryptionkey/current.go [hunk 1]
```diff
@@ -2,8 +2,32 @@ package encryptionkey
...
 )
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	secretName := getSecretName(customObject.Spec.Guest.ID)
```
Comment (xh3b4sd) [inline service/kvmclusterconfig/v1/resource/encryptionkey/current.go]: Lets please use the key package and have functions in there.

File: service/kvmclusterconfig/v1/resource/encryptionkey/current.go [hunk 2]
```diff
@@ -2,8 +2,32 @@ package encryptionkey
...
+
+	secret, err := r.k8sClient.Core().Secrets(v1.NamespaceDefault).Get(secretName, apismetav1.GetOptions{})
+
+	if apierrors.IsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return secret, nil
```
Comment (xh3b4sd) [inline service/kvmclusterconfig/v1/resource/encryptionkey/current.go]: Lets please stick to the pattern of having logs before and after the action.

File: service/kvmclusterconfig/v1/resource/encryptionkey/current_test.go [hunk 3]
```diff
@@ -1 +1,200 @@
...
+	}
+
+	for i, tc := range testCases {
+		objs := make([]runtime.Object, 0, len(tc.PresentSecrets))
+		for _, s := range tc.PresentSecrets {
+			objs = append(objs, s)
+		}
+
+		client := fake.NewSimpleClientset(objs...)
+		client.ReactionChain = append(tc.APIReactors, client.ReactionChain...)
```
Comment (xh3b4sd) [inline service/kvmclusterconfig/v1/resource/encryptionkey/current_test.go]: How does the reaction chain work?

File: service/kvmclusterconfig/v1/resource/encryptionkey/current.go (hunk 1 above)
Comment (tuommaki) [reply service/kvmclusterconfig/v1/resource/encryptionkey/current.go]: Ah, sure. I'll fix this.

File: service/kvmclusterconfig/v1/resource/encryptionkey/current.go (hunk 2 above)
Comment (tuommaki) [reply service/kvmclusterconfig/v1/resource/encryptionkey/current.go]: Will add logging there.

**Comment by tuommaki** [comment]
Tested on `geckon`.

**Comment by tuommaki** [comment]
@xh3b4sd You're good here?

File: service/kvmclusterconfig/v1/key/key.go [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+	"fmt"
+
 	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/randomkeytpr"
 )
 
+// ToCustomObject converts value to v1alpha1.KVMClusterConfig and returns it or
+// error if type does not match.
 func ToCustomObject(v interface{}) (v1alpha1.KVMClusterConfig, error) {
```
Comment (xh3b4sd) [inline service/kvmclusterconfig/v1/key/key.go]: I would like to see the functions being sorted alphabetically like in the other projects.

**Pull request merged**